### ISSUE TITLE
Method `search` has an interface problem

### ISSUE THREAD

**Issue by chainyo** [nephi-dev/rxml#3]
Hi, I face a problem while using the `node.search` method.

* This works but it's not great:

```python
node.search(SearchType.Tag, "...")
```

It works but mypy is warning me about the fact that `SearchType` doesn't have a `Tag` attribute, which is tru as everything is lowercased in the python Enum.

* This fails but should work according to the `search` method type hints:

```python
node.search("tag", "...")
# OR
node.search("Tag", "...")
```

This fails as the string can't be converted to a SearchType (`TypeError: argument 'by': 'str' object cannot be converted to 'SearchType'`)

I think the problem comes from the way the `SearchType` Enum is defined in the `rxml.pyi` file.

According to this [Single Enum section](https://pyo3.rs/v0.23.3/class.html?highlight=enum#pyclass-enums) of the `pyo3` documentation the python SerachType class should be like this:

```python
class SearchType:
    Tag: 'SearchType'  # These are class attributes
    Attr: 'SearchType'
    Text: 'SearchType'
    
    def __eq__(self, other: object) -> bool: ...
    def __int__(self) -> int: ...
```

And the Node search method should look like this:

```python
def search(self, by: SearchType, value: str, depth: int | None = None) -> list[Node]: ...
```

**Comment by nephi-dev** [comment]
sry, i've really written the pyi wrongly, the implementation does use uppercase on the first letter(as you can see [here](https://github.com/nephi-dev/rxml/blob/main/src/entities.rs#L6-L9)), i will fix it

**Comment by nephi-dev** [comment]
same for the search method definition, i've written it wrong, fixed on the [2.1.2 release](https://github.com/nephi-dev/rxml/releases/tag/2.1.2)

**Comment by nephi-dev** [comment]
actually, by default python(and pyo3) enum are integers, so, i'm changing the def to int on the pyi, releasing as 2.1.3

2.2.0 for newer version of all libs
